src/main/java/ch/uzh/ifi/hase/soprafs24/service/AIService.java

```diff
@@ -0,0 +1,67 @@
+package ch.uzh.ifi.hase.soprafs24.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.*;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.BodyInserters;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ResponseStatusException;
+import reactor.core.publisher.Mono;
+
+@Service
+public class AIService {
+  private final WebClient webClient;
+  private final ObjectMapper objectMapper;
+
+  @Value("${OPENAI_API_KEY}") private String openAiApiKey;
+
+  // Constructor for the service with WebClient and ObjectMapper
+  public AIService(WebClient.Builder webClientBuilder, ObjectMapper objectMapper) {
+    this.webClient = webClientBuilder.baseUrl("https://api.openai.com/v1").build();
+    this.objectMapper = objectMapper;
+  }
+
+  // Method to generate a description from a provided prompt
+  public Optional<String> generateDescription(String prompt) {
+    try {
+      String requestBody = createRequestBody(prompt);
+      String responseBody = sendPostRequest(requestBody);
+      String description = parseResponseBody(responseBody);
+      return Optional.of(description);
+    } catch (Exception e) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "AI Service failed");
+    }
+  }
+
+  // Method to create a request body for the POST request
+  private String createRequestBody(String prompt) {
+    return String.format(
+        "{\"model\": \"gpt-3.5-turbo-instruct\", \"prompt\":\"%s\", \"max_tokens\": 50}", prompt);
+  }
+
+  // Method to send the POST request to the OpenAI API
+  private String sendPostRequest(String requestBody) {
+    Mono<String> response = webClient.post()
+                                .uri("/completions")
+                                .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                                .header(HttpHeaders.AUTHORIZATION, "Bearer " + openAiApiKey)
+                                .body(BodyInserters.fromValue(requestBody))
+                                .retrieve()
+                                .onStatus(HttpStatus::is4xxClientError,
+                                    clientResponse -> Mono.error(new Exception("Client Error")))
+                                .onStatus(HttpStatus::is5xxServerError,
+                                    clientResponse -> Mono.error(new Exception("Server Error")))
```

Comment:
please throw `new ResponseStatusException(...)` with an exception that makes sense